packages/rdom/src/behavior/node.rs

```
@@ -25,5 +25,5 @@ pub trait NodeBehavior {
     /// [Node.getType](https://developer.mozilla.org/en-US/docs/Web/API/Node/nodeType)
     fn get_node_type(&self) -> isize;
     /// [.querySelector](https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelector)
-    fn query_selector(&self, selector: &Selector) -> Option<ElementNodeArc>;
+    fn query_selector(&self, selector: &Selector) -> Result<Option<ElementNodeArc>, DomError>;
```

This API seems a bit clumsy to use because of the double `.unwrap()`. Maybe we could add the `None` option into the error type somehow...

---

I agree it’s a little awkward. I think web_sys might even be like this in places. One way we could make it nicer is there is a plan to create a facade module that will provide an alternative interface to the library; it could panic instead of returning a DomError if the sandbox was dropped.